### DISCUSSION TITLE
Impact of strict and silent parameters in tar_load

### DISCUSSION THREAD

**Discussion by agila5** [ropensci/targets#816] [Help] [answered]
Dear all, I have a (probably trivial) question regarding the effect of the `strict` and `silent` parameters in `tar_load()`. From the help page I 
read: 

> `strict`: Logical of length 1, whether to error out if one of the selected targets cannot be loaded. Set to `FALSE` to just load the targets that can be loaded and skip the others.
> `silent`: Logical of length 1. If `silent` is `FALSE` and `strict` is `FALSE`, then a message will be printed if a target cannot be loaded, but load failures will not stop other targets from being loaded.

The default values are `strict = TRUE` and `silent = FALSE` and I would expect the following call to `tar_load()` to error out with a message: 

``` r
library(targets)
tar_dir({
  tar_script({
    list(
      tar_target(y1, 1 + 1),
      tar_target(y2, 1 + 1),
      tar_target(z, y1 + y2)
    )
  }, 
  ask = FALSE
  )
  tar_make()
  tar_load(c("A", "y1")) # An error here, no? 
  cat("the end!\n")
})
#> * start target y1
#> * built target y1
#> * start target y2
#> * built target y2
#> * start target z
#> * built target z
#> * end pipeline: 0.07 seconds
#> the end!
```

<sup>Created on 2022-04-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Why there is no error and no message?

**Comment by wlandau** [comment, marked as answer]
`tar_load()` ignores targets not in the metadata (like `A`). Errors can happen when you delete an object from `_targets/objects/` (with `tar_delete()` but the metadata is still there. 8c14325789ad27e6ec2746f0138fb78bc69128e6 addresses this finer point in the docs.

**Reply by agila5** [reply]
Thanks for the clarification!
